applications/ParticleMechanicsApplication/custom_strategies/schemes/mpm_explicit_scheme.hpp

```diff
@@ -0,0 +1,626 @@
+
+            //***************************************************************************
+            //***************************************************************************
+
+            void UpdateTranslationalDegreesOfFreedom(
+                const ProcessInfo& r_current_process_info,
+                NodeIterator itCurrentNode,
+                const IndexType DisplacementPosition,
+                const double delta_time,
+                const SizeType DomainSize = 3
+                )
+            {
+                std::array<bool, 3> fix_displacements = { false, false, false };
+                fix_displacements[0] = (itCurrentNode->GetDof(DISPLACEMENT_X, DisplacementPosition).IsFixed());
+                fix_displacements[1] = (itCurrentNode->GetDof(DISPLACEMENT_Y, DisplacementPosition + 1).IsFixed());
+                if (DomainSize == 3)
+                    fix_displacements[2] = (itCurrentNode->GetDof(DISPLACEMENT_Z, DisplacementPosition + 2).IsFixed());
+
+                array_1d<double, 3>& r_nodal_momenta = itCurrentNode->FastGetSolutionStepValue(NODAL_MOMENTUM);
+                array_1d<double, 3>& r_current_residual = itCurrentNode->FastGetSolutionStepValue(FORCE_RESIDUAL);
+
+                const double gamma = (r_current_process_info.GetValue(IS_EXPLICIT_CENTRAL_DIFFERENCE))
+                    ? 0.5
+                    : 1.0;
+                    // we are only adding the central difference corrector here
+
+                for (IndexType j = 0; j < DomainSize; j++) 
+                {
+                    if (fix_displacements[j]) 
+                    {
+                        r_nodal_momenta[j] = 0.0;
+                        r_current_residual[j] = 0.0;
+                    }
+                    else {
+                        r_nodal_momenta[j] += gamma * delta_time * r_current_residual[j];
+                    }
+                } // for DomainSize
+
+                // We need to set updated grid velocity here if we are using the USL formulation
+                if (r_current_process_info.GetValue(EXPLICIT_STRESS_UPDATE_OPTION) == 1)
+                {
+                    array_1d<double, 3>& r_current_velocity = itCurrentNode->FastGetSolutionStepValue(VELOCITY);
+                    r_current_velocity.clear();
+                    const double nodal_mass = itCurrentNode->FastGetSolutionStepValue(NODAL_MASS);
+                    if (nodal_mass > numerical_limit)
+                    {
+                        for (IndexType j = 0; j < DomainSize; j++)
+                        {
+                            r_current_velocity[j] = r_nodal_momenta[j] / nodal_mass;
+                        } // for DomainSize
+                    }
+                }
+            }
+
+            
+            /** This is the place to initialize the elements.
+            This is intended to be called just once when the strategy is initialized */
+            void InitializeElements(ModelPart& rModelPart) override
```

Comment:
This seems to do exactly the same as the baseclass
Or am I wrong?

Reply:
Nah, you're right - removed